SilverSolver/assignment3/word_search.py

```diff
@@ -0,0 +1,49 @@
+from bisect import bisect_left
+from dictionary import Dictionary
+from typing import List, Tuple
+
+def empty_matrix(m: int, n: int, value: bool) -> List[List[bool]]:
+    # value should has immutable type!
+    result = []
+    for i in range(m):
+        result.append([value] * n)
+    return result
+
+def word_search(grid: List[List[str]], dictionary: Dictionary) -> List[str]: 
+    # -- Should we check grid correctness?
```

Comment:
It is really good for the interviewer to get to know if the candidate thought about validation of the input. Some candidate would forget about it. Just clearly mention your assumption that the grid is correct, and that your priority is results, concision and clarity in the context of interviews. You can mention to the interviewer that you will implement that method iff there is time left.